### PR TITLE
C588 Check in: at service point not assigned to item's effective location (vega) (TaaS)

### PR DESCRIPTION
**Pull request by ernestyedigaryan1**
[C588](https://foliotest.testrail.io/index.php?/cases/view/588)

[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4440/allure/)

### CODE REVIEW

File: cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import getRandomPostfix from '../../support/utils/stringTools';
+import CheckInActions from '../../support/fragments/check-in-actions/checkInActions';
+import CheckInPane from '../../support/fragments/check-in-actions/checkInPane';
+import InTransit from '../../support/fragments/checkin/modals/inTransit';
+import ServicePoints from '../../support/fragments/settings/tenant/servicePoints/servicePoints';
+import Location from '../../support/fragments/settings/tenant/locations/newLocation';
+import InventoryInstances from '../../support/fragments/inventory/inventoryInstances';
+import InventoryInstance from '../../support/fragments/inventory/inventoryInstance';
+import PatronGroups from '../../support/fragments/settings/users/patronGroups';
+import permissions from '../../support/dictionary/permissions';
```
Comment (dnovikau) [inline cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js:14]: this import is redundant

**Review by manowarqqq — APPROVED** [review]

**Review by dnovikau — CHANGES_REQUESTED** [review]
I'd recommend to refactor the test by using newly added function for generation and creation of folio instances.

Example: cypress/e2e/loans/export-loans.cy.js

File: cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          itemData.holdingId = specialInstanceIds.holdingIds[0].id;
+          itemData.itemId = specialInstanceIds.holdingIds[0].itemIds;
+        });
+      });
+
+    PatronGroups.createViaApi(patronGroup.name).then((patronGroupResponse) => {
+      patronGroup.id = patronGroupResponse;
+    });
+
+    cy.createTempUser([Permissions.checkinAll.gui, Permissions.loansView.gui], patronGroup.name)
```
Comment (dnovikau) [inline cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js:92]: Permissions.loansView - is not mentioned in testcase

File: cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      testData.defaultLocation.campusId,
+      testData.defaultLocation.libraryId,
+      testData.defaultLocation.id,
+    );
+    cy.deleteLoanType(testData.loanTypeId);
+  });
+
+  it(
+    "C588 Check in: at service point not assigned to item's effective location (vega) (TaaS)",
+    { tags: [TestTypes.criticalPath, DevTeams.vega, Parallelization.nonParallel] },
```
Comment (dnovikau) [inline cypress/e2e/checkin/check-in-service-point-not-assigned.cy.js:134]: Is Parallelization.nonParallel required?

**Review by dnovikau — APPROVED** [review]

**Pull request merged**